core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/GatherAndMergeWindows.java

```diff
@@ -106,7 +106,7 @@ public Result apply(WindowNode parent, Captures captures, Context context)
             return pullWindowNodeAboveProjects(captures.get(childCapture), projects)
                     .flatMap(newChild -> manipulateAdjacentWindowNodes(parent, newChild, context))
                     .map(Result::ofPlanNode)
-                    .orElse(Result.empty());
```

Comment:
Result.empty() should be very cheap. It might even be cheaper than creating the lambda expression

Reply:
in this case the lambda is constant, so it's not being created.

Reply:
The ref is in my head, so it may not be valid, but i believe constant lambda gets converted to a constant, so it ends up being `orElseGet(const)` call.

Anyway, let me roll this back. The `Result.empty()` should be cheap.